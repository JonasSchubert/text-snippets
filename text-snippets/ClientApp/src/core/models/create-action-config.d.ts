import { ActionTree } from "vuex";
import { DefaultState } from "./default-state";
import { Entity } from "./entity";
import { RootState } from "./root-state";

export interface CreateActionConfig<T extends Entity> {
  actions?: ActionTree<DefaultState<T>, RootState>;
  route: string;
  type: string;
}
