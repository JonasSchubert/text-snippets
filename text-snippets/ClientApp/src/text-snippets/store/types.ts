export const ActionTypes = {
  add: 'add',
  delete: 'delete',
  load: 'load',
  update: 'update'
};

export const GetterTypes = {
  error: 'error',
  isLoading: 'isLoading',
  list: 'list'
};

export const ModuleType = 'text-snippets';

export const MutationTypes = {
  setError: 'setError',
  setIsLoading: 'setIsLoading',
  setList: 'setList'
};

export const RouteTypes = {
  textSnippets: 'TextSnippets'
};
