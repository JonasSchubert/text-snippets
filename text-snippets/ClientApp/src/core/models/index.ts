export * from './root-state.d';
