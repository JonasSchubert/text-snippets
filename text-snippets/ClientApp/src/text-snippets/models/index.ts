export * from './text-snippet.d';
export * from './text-snippets-state.d';
