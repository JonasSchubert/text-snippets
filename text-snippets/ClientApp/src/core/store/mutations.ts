import { MutationTree } from 'vuex';
import { RootState } from '@/core/models';
import { MutationTypes } from './types';

export const createMutations = (): MutationTree<RootState> => ({
  [MutationTypes.decreaseRequests]: (state: RootState): void => { state.requests -= 1; },
  [MutationTypes.increaseRequests]: (state: RootState): void => { state.requests += 1; }
});
