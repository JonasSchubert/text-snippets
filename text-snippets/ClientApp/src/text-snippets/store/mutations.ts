import { MutationTree } from 'vuex';
import { TextSnippet, TextSnippetsState } from '@/text-snippets/models';
import { MutationTypes } from './types';

export const createMutations = (): MutationTree<TextSnippetsState> => ({
  [MutationTypes.setError]: (state: TextSnippetsState, { error }: { error: Error }): void => { state.error = error; },
  [MutationTypes.setIsLoading]: (state: TextSnippetsState, { isLoading }: { isLoading: boolean }): void => { state.isLoading = isLoading; },
  [MutationTypes.setList]: (state: TextSnippetsState, { list }: { list: TextSnippet[] }): void => { state.list = list; }
});
