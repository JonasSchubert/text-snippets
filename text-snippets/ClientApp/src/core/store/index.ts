import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createModule as createTextSnippetsModule } from '@/text-snippets/store';
import { ModuleType as TextSnippetsModule } from '@/text-snippets/store/types';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('text-snippets', vuetify),
      [TextSnippetsModule]: createTextSnippetsModule()
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
