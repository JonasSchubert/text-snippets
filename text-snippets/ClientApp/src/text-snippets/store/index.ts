import { Module } from 'vuex';
import { RootState } from '@/core/models';
import { TextSnippetsState } from '@/text-snippets/models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = (): Module<TextSnippetsState, RootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(),
  mutations: createMutations(),
  state: createState()
});
