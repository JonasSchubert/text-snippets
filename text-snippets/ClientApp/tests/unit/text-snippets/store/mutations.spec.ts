import { TextSnippet, TextSnippetsState } from '@/text-snippets/models';
import { createMutations } from '@/text-snippets/store/mutations';
import { MutationTypes } from '@/text-snippets/store/types';

describe('text-snippets/mutations', () => {
  let state: TextSnippetsState;
  const mutations = createMutations();

  beforeEach(() => {
    state = {
      error: undefined,
      isLoading: false,
      list: []
    };
  });

  describe(`${MutationTypes.setError}`, () => {
    test('should set error', () => {
      // Arrange
      const error: Error = {
        message: 'message',
        name: 'name'
      };

      // Act
      mutations[MutationTypes.setError](state, { error });

      // Assert
      expect(state.error).toMatchSnapshot({
        message: 'message',
        name: 'name'
      });
    });
  });

  describe(`${MutationTypes.setIsLoading}`, () => {
    test('should set isLoading', () => {
      // Arrange
      const isLoading = true;

      // Act
      mutations[MutationTypes.setIsLoading](state, { isLoading });

      // Assert
      expect(state.isLoading).toBeTruthy();
    });
  });

  describe(`${MutationTypes.setList}`, () => {
    test('should set list', () => {
      // Arrange
      const list: TextSnippet[] = [
        { id: '1' } as TextSnippet
      ];

      // Act
      mutations[MutationTypes.setList](state, { list });

      // Assert
      expect(state.list).toMatchSnapshot([
        { id: '1' }
      ]);
    });
  });
});
