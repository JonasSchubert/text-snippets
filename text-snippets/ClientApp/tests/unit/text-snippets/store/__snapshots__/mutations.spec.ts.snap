// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-snippets/mutations setError should set error 1`] = `
Object {
  "message": "message",
  "name": "name",
}
`;

exports[`text-snippets/mutations setList should set list 1`] = `
Object {
  "0": Object {
    "id": "1",
  },
}
`;
